src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs

```diff
@@ -0,0 +1,212 @@
+using System.Collections.Generic;
+using System.Data;
+using System.Text;
+using Microsoft.Data.SqlClient.ManualTesting.Tests.AlwaysEncrypted.Setup;
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.ManualTesting.Tests.AlwaysEncrypted
+{
+    [PlatformSpecific(TestPlatforms.Windows)]
+    public class SqlNullValuesTests : IClassFixture<SQLSetupStrategyCertStoreProvider>, IDisposable
+    {
+        private SQLSetupStrategyCertStoreProvider fixture;
+        private readonly string tableName;
+        private string UdfName = DatabaseHelper.GenerateUniqueName("SqlNullValuesRetVal");
+        private string UdfNameNotNull = DatabaseHelper.GenerateUniqueName("SqlNullValuesRetValNotNull");
+
+        public SqlNullValuesTests(SQLSetupStrategyCertStoreProvider fixture)
+        {
+            this.fixture = fixture;
+            tableName = fixture.SqlNullValuesTable.Name;
+            // Disable the cache to avoid false failures.
+            SqlConnection.ColumnEncryptionQueryMetadataCacheEnabled = false;
+
+            foreach (string connStr in DataTestUtility.AEConnStringsSetup)
+            {
+                // Insert data and create functions for SqlNullValues test.
+                using (SqlConnection sqlConnection = new SqlConnection(connStr))
+                {
+                    sqlConnection.Open();
+
+                    using (SqlCommand cmd = new SqlCommand(String.Format("INSERT INTO [{0}] (c1) VALUES (@c1)", tableName), sqlConnection, null, SqlCommandColumnEncryptionSetting.Enabled))
+                    {
+                        SqlParameter param = cmd.Parameters.Add("@c1", SqlDbType.Int);
+                        param.Value = DBNull.Value;
+                        cmd.ExecuteNonQuery();
+
+                        param.Value = 10;
+                        cmd.ExecuteNonQuery();
+                    }
+
+                    string sql1 = $"CREATE FUNCTION {UdfName}() RETURNS INT AS \n BEGIN \n RETURN (SELECT c1 FROM [{tableName}] WHERE c1 IS NULL)\n END";
+                    string sql2 = $"CREATE FUNCTION {UdfNameNotNull}() RETURNS INT AS \n BEGIN \n RETURN (SELECT c1 FROM [{tableName}] WHERE c1 IS NOT NULL)\n END";
+                    using (SqlCommand cmd = sqlConnection.CreateCommand())
+                    {
+                        cmd.CommandText = sql1;
+                        cmd.ExecuteNonQuery();
+
+                        cmd.CommandText = sql2;
+                        cmd.ExecuteNonQuery();
+                    }
+                }
+            }
+        }
+
+        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringSetupForAE))]
+        [ClassData(typeof(NullValueTestsData))]
+        public void NullValueTests(string connString, ConnectionStringSetting connStringSetting, SqlCommandColumnEncryptionSetting commandSetting, ReturnValueSetting nullReturnValue)
+        {
+            switch (connStringSetting)
+            {
+                case ConnectionStringSetting.Enabled:
```

Comment:
"ConnectionStringSetting" can be renamed to "ConnStringColumnEncryptionSetting"?
(Looking at SqlCommandColumnEncryptionSetting)

Reply:
Updated